Directory.Packages.props

```diff
@@ -6,7 +6,7 @@
     <NewtonsoftJsonPackageVersion Condition="'$(NewtonsoftJsonPackageVersion)' == ''">13.0.3</NewtonsoftJsonPackageVersion>
     <SystemTextJsonVersion Condition="'$(SystemTextJsonVersion)' == ''">7.0.3</SystemTextJsonVersion>
     <SystemPackagesVersion Condition="'$(SystemPackagesVersion)' == ''">4.3.0</SystemPackagesVersion>
-    <VSFrameworkVersion Condition="'$(VSFrameworkVersion)' == ''">17.8.37297</VSFrameworkVersion>
+    <VSFrameworkVersion Condition="'$(VSFrameworkVersion)' == ''">17.10.39524</VSFrameworkVersion>
```

Comment:
Further down the file, there are some project-dependent `PackageVersion` items, just for our VS Extensibility packages that depend on specific packages, rather than the whole VS SDK. It'd be good to update these packages to the same versions that this VS SDK meta package references:

https://github.com/NuGet/NuGet.Client/blob/de9c314080687bfe4da4baf4b3d1a9d38b6ace3a/Directory.Packages.props#L119-L135

Reply:
> These individual package versions should be equal to or less than the package version within the VS SDK that our
>     customers use, otherwise they might get NU1605 or MSB3277 warnings.

Do you know the answer to the question of which SDK version our customers use?

After changing these package versions to match the version used by the VS SDK, the first error I'm getting seems to want me to pick a version. I'm not sure if I pick the one the VS SDK references, or if there's some other lowest common denominator that customers would use.

> Severity	Code	Description	Project	File	Line	Suppression State	Details
> Error (active)	NU1603	Warning As Error: Microsoft.VisualStudio.ComponentModelHost 17.10.126-preview depends on Microsoft.VisualStudio.Interop (>= 17.10.39408-preview.3) but Microsoft.VisualStudio.Interop 17.10.39408-preview.3 was not found. An approximate best match of Microsoft.VisualStudio.Interop 17.10.39524 was resolved.	NuGet.VisualStudio.Client	C:\NuGet.Client\src\NuGet.Clients\NuGet.VisualStudio.Client\NuGet.VisualStudio.Client.csproj	1		
>

Reply:
It seems there's no way to satisfy the condition of

The VS SDK dependency, `"Microsoft.VisualStudio.ComponentModelHost" version="17.10.126-preview"`
...depends on Package  `Microsoft.VisualStudio.Imaging.Interop.14.0.DesignTime (>= 17.10.39365-preview.3)`
...but VS SDK depends on `"Microsoft.VisualStudio.Imaging.Interop.14.0.DesignTime" version="17.10.39500"`

Package `Microsoft.VisualStudio.ComponentModelHost 17.10.133-preview` is the version which also depends on `17.10.39500` as above.

If you know a way to reconcile this, let me know.

Reply:
I think this is where using public versions of the VS SDK published on nuget.org is helpful, since I think there's some additional process to increase the probability that the VS SDK is "coherent".

On nuget.org, there exists [version 17.10.40171 of the VS SDK](https://www.nuget.org/packages/Microsoft.VisualStudio.SDK/17.10.40171), which is higher than the version that this PR has been proposing.

This version of the VS SDK depends on [Microsoft.VisualStudio.ComponentModelHost 17.10.191](https://www.nuget.org/packages/Microsoft.VisualStudio.ComponentModelHost/17.10.191#dependencies-body-tab), which itself depends on Microsoft.VisualStudio.Imaging.Interop.14.0.DesignTime version 17.10.39546, while the VSSDK itself depends on version 17.10.40152, which is higher. Therefore, since the VS SDK is bringing in a higher version of that transitive dependency, I wouldn't expect any NU1603 (package downgrade) issues.

Honestly, I'm wondering if we should make NuGet.VisualStudio.Contracts depend on the VS SDK rather than individual packages. I'm no longer sure that the circular reference concern in the comments is correct. But this may not be feasible if we end up adding brokered services for the C# Dev Kit in order to share implementation between VSCode and VS. Plus it would mean that we couldn't/shouldn't ever use a pre-release VS SDK to integrate with VS features that are not yet GA. But having NuGet use a GA version of the VS SDK is compatible with this PR, since you're not trying to use a 17.11 VS SDK in this PR.

> Do you know the answer to the question of which SDK version our customers use?

I think that external customers are very unlikely to use the VSIDE feed, and probably only use the nuget.org packages. Since our dev branch is always targeting VS version vNext, by using the last stable (GA) version, we'll never release a version of NuGet.VisualStudio packages that have "bad" dependencies, since our package will always depend on the previous minor version of the VS SDK, not the same release.

Reply:
I pushed changes to use the versions of the packages actually on nuget.org, even if they aren't the same as what the VS SDK meta package depend on.